wps-cloudformation-template.yaml

```diff
@@ -57,7 +57,44 @@ Parameters:
   geoserverWpsEndpoint:
     Type: String
     Default: 'http://geoserver-wps.aodn.org.au/geoserver/wps'
+  environment:
+    Type: String
+    Default: dev
+    AllowedValues:
+      - production
+      - systest
+      - sandbox
+      - dev
+  createProdSystestBucket:
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+Mappings:
+  EnvironmentMap:
+    production:
+      BucketName: "aws-wps-production"
+    systest:
+      BucketName: "aws-wps-systest"
+Conditions:
+  CreateProdSystestBucket: !And
+    - !Equals [!Ref createProdSystestBucket, true]
+    - !Not [!Condition CreateSandboxDevBucket]
+  CreateSandboxDevBucket: !Not [!Or [!Equals [ !Ref environment, production ], !Equals [ !Ref environment, systest ]]]
 Resources:
+  S3BucketProdSystest:
+    Type: "AWS::S3::Bucket"
+    Condition: CreateProdSystestBucket
+    DeletionPolicy: "Retain"
+    Properties:
+      BucketName: !FindInMap [EnvironmentMap, !Ref environment, BucketName]
```

Comment:
See above re: generating bucket name from environment name rather than needing a map.

Reply:
It makes sense to dynamically allocate a bucket name to ephemeral environments. But in case of production or systest we will have to use a fixed bucket name.